src/PanZoom.js

```diff
@@ -240,15 +310,52 @@ class PanZoom extends React.Component<Props, State> {
   }
 
   onWheel = (e: WheelEvent) => {
-    const { disableScrollZoom, disabled, zoomSpeed } = this.props
-    if (disableScrollZoom || disabled) {
-      return
-    }
+    const { disableScrollZoom, disabled, zoomSpeed, noStateUpdate, hasNaturalScroll } = this.props
 
-    const scale = getScaleMultiplier(e.deltaY, zoomSpeed)
-    const offset = this.getOffset(e)
-    this.zoomTo(offset.x, offset.y, scale)
-    e.preventDefault()
+    if (disabled) return
+
+    if (!disableScrollZoom && this.ctrlKeyPressed) {
```

Comment:
This first section handles the zoom functionality while panning. If the control key is pressed down and wheel events are happening, then canvas will zoom in or out.